yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -61,6 +60,12 @@ private[spark] class YarnClientSchedulerBackend(
     // to the executors
     super.start()
 
+    // SPARK-8851: In yarn-client mode, the AM still does the credentials refresh. The driver
+    // reads the credentials from HDFS, just like the executors and updates its own credentials
+    // cache.
+    if (conf.contains("spark.yarn.credentials.file")) {
+      YarnSparkHadoopUtil.get.startExecutorDelegationTokenRenewer(conf)
+    }
     waitForApplication()
```

Comment:
Will do. This might not be that big an issue since the app is likely to be submitted and `waitForApplication` returned before we need the tokens to be updated (wouldn't this cause bigger issues otherwise - like the executors might not be able to start/read HDFS data because the tokens already expired etc?). Basically this just means that the token expiry interval is too short to be useful/